pandas/tests/io/test_sql.py

```diff
@@ -741,6 +741,35 @@ def test_date_parsing(self):
             Timestamp("2010-12-12"),
         ]
 
+    @pytest.mark.parametrize("error", ["ignore", "raise", "coerce"])
+    @pytest.mark.parametrize(
+        "read_sql, text, mode",
+        [
+            (sql.read_sql, "SELECT * FROM types_test_data", ("sqlalchemy", "fallback")),
+            (sql.read_sql, "types_test_data", ("sqlalchemy")),
+            (
+                sql.read_sql_query,
+                "SELECT * FROM types_test_data",
+                ("sqlalchemy", "fallback"),
+            ),
+            (sql.read_sql_table, "types_test_data", ("sqlalchemy")),
+        ],
+    )
+    def test_custom_dateparsing_error(self, read_sql, text, mode, error):
+        if self.mode in mode:
+            df = read_sql(
+                text,
+                con=self.conn,
+                parse_dates={
+                    "DateCol": {"errors": error},
+                },
+            )
+            assert issubclass(df.DateCol.dtype.type, np.datetime64)
```

Comment:
Could you test the full expected dataframe with `tm.assert_frame_equal`?

Reply:
Np to do this, but all the other tests that use 'types_test_data' do not use `tm.assert_frame_equal`, but the assert approach I used. So just doublechecking if we want to go ahead with this.

Reply:
`assert_frame_equal` is preferable in general since it tests the asserts you have and more

Reply:
Updated the test, but sometimes the assert fails since the machine casts the integer that is read from the SQL db to int32. Giving the error:
```
E           Attribute "dtype" are different
E           [left]:  int64
E           [right]: int32
```

I've been trying some stuff, but without success. Is there a way to let pandas decide which int type to use, so that we don't have this mismatch? Else we can provide the argument `check_dtype=False` to `assert_frame_equal`, but IMO that seems a bit too much.

Reply:
nvm, it looks like I got it

Reply:
Done